stream-rpc/src/websocket.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use diem_config::config::{RoleType, WebsocketStreamConfig};
+use diem_logger::debug;
+use diem_types::chain_id::ChainId;
+use std::sync::Arc;
+use storage_interface::DbReader;
+use tokio::{
+    runtime::{Builder, Runtime},
+    sync::mpsc,
+};
+use tokio_stream::wrappers::ReceiverStream;
+
+use warp::{filters::BoxedFilter, ws::Message, Filter, Rejection, Reply};
+
+use crate::{
+    connection::{ConnectionContext, ConnectionManager, Transport},
+    subscriptions::SubscriptionConfig,
+    counters, logging,
+};
+use futures::{future::Either, FutureExt, StreamExt};
+
+pub fn get_websocket_routes(config: &WebsocketStreamConfig, diem_db: Arc<dyn DbReader>) -> BoxedFilter<(impl Reply, )> {
+    let sub_config = Arc::new(SubscriptionConfig {
+        fetch_size: config.subscription_fetch_size as u64,
+        poll_interval_ms: config.poll_interval_ms as u64,
+        queue_size: config.send_queue_size as u64,
```

Comment:
You shouldn't need to use the `as` operator. I would instead make sure that these Configs use the same underlying types

Reply:
the `queue_size` is used in two places which each require a different type; I've reduced the number of casts but I'll still have to cast this one somewhere

Reply:
Where else is it used?